### ISSUE TITLE
Improvement: ProjectApi support `min_access_level` filter

### ISSUE THREAD

**Issue by annguyen-qh** [gitlab4j/gitlab4j-api#356]
It would be great if ProjectApi could support `min_access_level` [filter](https://docs.gitlab.com/ee/api/projects.html#list-all-projects) when listing projects accessible by the authenticated user. (ProjectApi.getProjects )

Currently, only ProjectApi.getUserProjects support this filter (via ProjectFilter)

**Comment by gmessner** [comment]
@annguyen-qh 
The ability to fetch projects with a filter has been added an released in 4.10.19. The new methods for getting projects using a filter in ```ProjectApi``` are:
```
public List<Project> getProjects(ProjectFilter filter)
public List<Project> getProjects(ProjectFilter filter, int page, int perPage)
public Pager<Project> getProjects(ProjectFilter filter, int itemsPerPage)
public Stream<Project> getProjectsStream(ProjectFilter filter)
```
In addition to the ```min _access_level``` filter you can also filter on some recently added filter attributes:
```
with_programming_language
wiki_checksum_failed
repository_checksum_failed
```
If you could verify and close this issue I would appreciate it.

**Comment by annguyen-qh** [comment]
Hi @gmessner,
Thank you for the update. `ProjectFilter` works fine except `min_access_level` field. 

The generated url looks like this: `https://gitlab.com/api/v4/projects?min_access_level+=40`. I guess `+` sign comes from the extra space in `"min_access_level "` string

```
.withParam("min_access_level ", (minAccessLevel != null ? minAccessLevel.toValue() : null)
```

**Comment by gmessner** [comment]
@annguyen-qh 
That was the issue, thanks for identifying it.  I fixed this and added a test to ensure that the ```min_access_level``` filter is working correctly.  The fix is in release 4.10.20 which was just pushed to maven central.

**Comment by annguyen-qh** [comment]
Hi @gmessner ,
I’ve upgraded to the latest version (4.11.0) and verified.
Thank you for the quick fix.
